### ISSUE TITLE
chawan: `mancha -k` doesn't find results

### ISSUE THREAD

**Issue by eclairevoyant** [NixOS/nixpkgs#346082]
### Describe the bug
`mancha -k` doesn't find results

### Steps To Reproduce
```
$ mancha -k regular
cha: Can't load man-k:regular (nothing appropriate)
```

### Additional context
`man -k` works:

```
$ man -k regular   
bzegrep (1)          - search possibly bzip2 compressed files for a regular expression
bzfgrep (1)          - search possibly bzip2 compressed files for a regular expression
bzgrep (1)           - search possibly bzip2 compressed files for a regular expression
pam_usertype (8)     - check if the authenticated user is a system or regular account
perlfaq6 (1)         - Regular Expressions
perlre (1)           - Perl regular expressions
perlreapi (1)        - Perl regular expression plugin interface
perlrebackslash (1)  - Perl Regular Expression Backslash Sequences and Escapes
perlrecharclass (1)  - Perl Regular Expression Character Classes
perlreguts (1)       - Description of the Perl regular expression engine.
perlrequick (1)      - Perl regular expressions quick start
perlreref (1)        - Perl Regular Expressions Reference
perlretut (1)        - Perl regular expressions tutorial
regex (7)            - POSIX.2 regular expressions
zgrep (1)            - search possibly compressed files for a regular expression
```

### Notify maintainers
@jtbx

### Metadata
I'm using `chawan` from `nixos-unstable`.
I even tried bumping `chawan` to the current upstream commit.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by rnhmjoj** [comment]
Is this thing reusing the man-db caches to search for man pages? If so, it probably can't find NixOS caches, man-db as well had to be patched to work correctly.

**Comment by jtbx** [comment]
If I recall correctly it's patched to use mandoc because of a particular flag mancha requires.

**Comment by bptato** [comment]
> Is this thing reusing the man-db caches to search for man pages? If so, it probably can't find NixOS caches, man-db as well had to be patched to work correctly.

No, it's nowhere that sophisticated. mancha just shells out to man, so if man works, then so should mancha.

> If I recall correctly it's patched to use mandoc because of a particular flag mancha requires.

It works with man-db on Debian, so I don't understand why this is needed.

**Comment by eclairevoyant** [comment]
> I don't understand why this is needed.

It's not directly mancha's fault, it's due to a patch added in nixpkgs which uses the `-m` flag to allow the nixpkgs build of `mancha` to find its own manpages, so `MANCHA_MAN` was also set to point to `mandoc`

**Comment by bptato** [comment]
Hmm. But doesn't that mean that `man cha` with the system man still
won't work?

At least I remember OpenBSD's `man` wouldn't find man pages in their
default installation directory, so I had to adjust MANPREFIX
manually. Perhaps this is a similar case?

**Comment by eclairevoyant** [comment]
> But doesn't that mean that `man cha` with the system man still won't work?

It seems to work. I'm not entirely sure about the differences with `mancha` and `man-db`.

**Comment by bptato** [comment]
No idea then :/
FWIW, mancha (the "new" version, from March 2024) calls man like this:

* if MANCHA_MAN is set, use it
* look for /usr/bin/man, /bin/man, /usr/local/bin/man (in this order)
* if found, use that; otherwise, run /usr/bin/env man

Unless this process resolves to a different `man` binary than what the shell normally finds in PATH, I don't see why it wouldn't work.

By the way, the "old" Perl version (which, it seems, the patch was originally written for) had unconditionally used /usr/bin/man.  Perhaps the "new" version would exhibit a different behavior without the patch?
